testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/persistence/EncodingPersistenceTestCase.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import jakarta.inject.Inject;
+import org.apache.commons.io.FileUtils;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildFlyRunner;
+
+@RunWith(WildFlyRunner.class)
+@ServerControl(manual = true)
+public class EncodingPersistenceTestCase {
+    static final PathAddress PROPERTY_ADDR = PathAddress.pathAddress(ModelDescriptionConstants.SYSTEM_PROPERTY, "test_property");
+    @Inject
+    private ServerController serverController;
+
+    /**
+     * Tests that we can write and read the persistence XML when the server is launched using an encoding different that UTF-8
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testEncodingManagementClient() throws Exception {
+        String originalArgs = System.getProperty("jboss.args");
+        try {
+            System.setProperty("jvm.args", originalArgs + " -Dfile.encoding=windows-1250");
+            serverController.start();
+
+            final ManagementClient client = serverController.getClient();
+
+            ModelNode op = Util.createAddOperation(PROPERTY_ADDR);
+            op.get(ModelDescriptionConstants.VALUE).set("áéíóú");
+
+            client.executeForResult(op);
+
+            serverController.stop();
+            serverController.start();
+        } finally {
+            System.setProperty("jboss.args", originalArgs);
+            if (serverController.isStarted()) {
+                serverController.stop();
+            }
+        }
+    }
+
+    /**
+     * Verifies all the persistence XML files explicitly have an encoding header as UTF-8
+     *
+     * @throws IOException
+     */
+    @Test
+    public void checkUtf8HeaderOnXmlConfigurationFiles() throws IOException {
+        final Path jbossHome = Paths.get(TestSuiteEnvironment.getJBossHome());
+
+        final Path standalone = jbossHome.resolve("standalone").resolve("configuration");
+        try (Stream<Path> file = Files.list(standalone)) {
+            file.filter(p -> p.toString().endsWith(".xml")).forEach(this::checkXmlHeader);
+        }
+
+        final Path domain = jbossHome.resolve("domain").resolve("configuration");
+        try (Stream<Path> file = Files.list(domain)) {
+            file.filter(p -> p.toString().endsWith(".xml")).forEach(this::checkXmlHeader);
+        }
```

Comment:
This block needs to be guarded with a Files.exists check.

Reply:
@bstansberry Fixed, although I'm still getting failures when `org.wildfly.core:wildfly-jmx:20.0.0.Beta5-SNAPSHOT`is being resolved; I'm now running out of ideas about what could cause the issue. 

@luck3y can you give some feedback about this? It is weird getting a build complaining about an artifact that has been installed into the maven cache before running this test.

Reply:
Notice the server for this test case is launched with a different encoding, but I've tested that this is not the reason for not being able to resolve the `org.wildfly.core:wildfly-jmx:20.0.0.Beta5-SNAPSHOT` from the local maven cache